### PR TITLE
Bug1529555 5.6

### PR DESCRIPTION
**Pull request by laurynas-biveinis**
```
http://jenkins.percona.com/job/percona-server-5.6-param/1070/
```

### CODE REVIEW

File: storage/innobase/handler/handler0alter.cc [hunk 1]
```diff
@@ -5886,6 +5886,21 @@ ha_innobase::commit_inplace_alter_table(
 	row_mysql_unlock_data_dictionary(trx);
 	trx_free_for_mysql(trx);
 
+	/* Rebuild index translation table now for temporary tables if we are
+	restoring secondary keys, as ha_innobase::open will not be called for
+	the next access.  */
+	if (DICT_TF2_FLAG_IS_SET(ctx0->new_table, DICT_TF2_TEMPORARY)
```
Comment (kbauskar) [inline storage/innobase/handler/handler0alter.cc]: Can we replace this with dict_table_is_temporary() to make it more readable and consistent with 5.5 too.

File: storage/innobase/handler/handler0alter.cc [hunk 2]
```diff
@@ -5886,6 +5886,21 @@ ha_innobase::commit_inplace_alter_table(
...
+	    && ctx0->num_to_add_index) {
```
Comment (kbauskar) [inline storage/innobase/handler/handler0alter.cc]: Optional but 5.7 we are enforcing rule that only boolean should be used w/o RHS.
Should if possible to make the condition more clear mentioning it as

(ctx0->num_of_add_index > 0)

As this is not a official rule will leave decision to developer.

**Comment by kbauskar** [comment]
LGTM. 2 trivial comments to address.

**Comment by laurynas-biveinis** [comment]
Comments addressed
http://jenkins.percona.com/job/percona-server-5.6-param/1073/

**Pull request merged**